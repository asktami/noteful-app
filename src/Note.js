import React, { useContext } from 'react';
import PropTypes from 'prop-types';
import NoteItem from './NoteItem';

import NotefulContext from './NotefulContext';

const Note = props => {
	// need to grab NotefulContext (globals)
	const context = useContext(NotefulContext);
	const { notes } = context;

	// NOTE
	// reloading the browser wipes the context from memory
	// so need to handle if user reloads the browser and its undefined
	// done by adding || {} here AND : '' for props.modified in NoteItem too
	const note = notes.find(note => note.id === props.match.params.noteId) || {};

	return (
		<article>
			<div className="note">
				<NoteItem {...props} note={note} />
				<p className="note-content">{note.content}</p>
			</div>
		</article>
	);
};

export default Note;

// to catch bugs
// check that get a notes array that has id, name, content, and modified
// this array is the "notes" variable coming from context
Note.propTypes = {
	notes: PropTypes.arrayOf(
		PropTypes.shape({
			id: PropTypes.string.isRequired,
			name: PropTypes.string.isRequired,
			content: PropTypes.string.isRequired,
			modified: PropTypes.string.isRequired
		})
	)
};
