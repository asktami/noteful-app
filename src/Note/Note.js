import React, { useContext } from 'react';
import PropTypes from 'prop-types';
import NoteItem from '../NoteItem/NoteItem';

import NotefulContext from '../NotefulContext';

const Note = props => {
	// need to grab NotefulContext (globals)
	const context = useContext(NotefulContext);
	const { notes } = context;

	// NOTE
	// reloading the browser wipes the context from memory
	// so need to handle if user reloads the browser and its undefined
	// done by adding || {} here AND : '' for props.modified in NoteItem too
	const note = notes.find(note => note.id === props.match.params.noteId) || {};

	return (
		<section>
			<div className="note">
				<NoteItem {...props} note={note} />
				<p className="note-content">{note.content}</p>
			</div>
		</section>
	);
};

export default Note;

// to catch bugs
// check that get a notes array that has id, name, content, and modified
// this array is the "notes" variable coming from context
Note.propTypes = {
	notes: PropTypes.arrayOf(
		PropTypes.shape({
			id: PropTypes.string.isRequired,
			folderId: PropTypes.string.isRequired,
			name: PropTypes.string.isRequired,
			content: PropTypes.string.isRequired,
			modified: PropTypes.instanceOf(Date).isRequired
		})
	)
};
