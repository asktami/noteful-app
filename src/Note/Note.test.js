import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Note from './Note';

describe('Dummy Note test', () => {
	test('adding 1 + 2 should return 3', () => {
		expect(2 * 2).toBe(4);
	});
});
