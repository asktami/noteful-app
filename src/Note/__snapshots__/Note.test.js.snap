// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Note component renders a Note by default 1`] = `
<section>
  <div
    className="note"
  >
    <NoteItem
      match={
        Object {
          "params": Object {
            "noteId": "1",
          },
        }
      }
      note={Object {}}
      notes={
        Array [
          Object {
            "content": "test-note-content",
            "folderId": "101",
            "id": "1",
            "modified": 2019-09-30T04:00:00.000Z,
            "name": "test-note-title",
          },
        ]
      }
    />
    <p
      className="note-content"
    />
  </div>
</section>
`;

exports[`Note component renders the Note given props 1`] = `
<section>
  <div
    className="note"
  >
    <NoteItem
      match={
        Object {
          "params": Object {
            "noteId": "1",
          },
        }
      }
      note={Object {}}
      notes={
        Array [
          Object {
            "content": "test-note-content",
            "folderId": "101",
            "id": "1",
            "modified": 2019-09-30T04:00:00.000Z,
            "name": "test-note-title",
          },
        ]
      }
    />
    <p
      className="note-content"
    />
  </div>
</section>
`;
